### ISSUE TITLE
Improve config validation and editing

### ISSUE THREAD

**Issue by thomasdarimont** [adorsys/keycloak-config-cli#1122]
### Problem Statement

Currently keycloak-config-cli can detect configuration errors like missing or unknown attribute only when running against keycloak. Also when editing config files, user have to know upfront which properties are supported which is errorprone and time consuming.

### Proposed Solution

It would be helpful if there were some schema support to help with the validation during editing. The schema could also be used to provide code completion in IDEs (vscode and intellij).

One way to implement this is to extract the schemes for the representations from the https://www.keycloak.org/docs-api/latest/rest-api/openapi.json OpenAPI spec and use it to validate the configuration file.

### Environment

- Keycloak Version: [e.g. 24.0.1]
- keycloak-config-cli Version: [e.g. 5.11.1]
- Java Version: [e.g. 21]

### Acceptance Criteria

YAML validation with JSON Schema in IntelliJ: https://www.jetbrains.com/help/idea/yaml.html#select-schema-for-file
YAML validation with JSON Schema in vscode: https://dev.to/brpaz/how-to-create-your-own-auto-completion-for-json-and-yaml-files-on-vs-code-with-the-help-of-json-schema-k1i

**Comment by Motouom** [comment]
Hello @thomasdarimont.

Did you do any progess on this ticket or should we work on this. If yes, we will staty reseaching on how to infer a json schema from an openapi file.

**Comment by thomasdarimont** [comment]
I didn't have time to work on this.
Feel free to give this a spin!

**Comment by Motouom** [comment]
Hello @thomasdarimont 

I have already done most of the validations config and editing, I just have a few other validations(specific validations to add.  I wish to ask if you have any idea on wether we can do it in such a way that, once the user opens the JSON or yaml config file, it should automatically be set to the JSON schema that is meant to kc-cli?

**Comment by thomasdarimont** [comment]
Thanks for picking this up. Regarding the config schema validation:
Have you tried this in intelliJ? https://www.jetbrains.com/help/idea/yaml.html#use-schema-keyword

**Comment by Motouom** [comment]
Yes, i have tried it already and it is working,  Even on vs code.


i was wondering if there was a way we could integrate the schema in the IDE other than  the way it is done in IntelliJ by configuring it.

for example:
- creating a URL for the json schema that can be accessible to everyone.
- creating some sort of an extension. 

i would like to have you thoughts on it.

**Comment by Motouom** [comment]
Hello @thomasdarimont, @Thendo20 , @AssahBismarkabah  please can you review this #1331? I will close the last PR(#1268).

**Comment by emouty** [comment]
Hello, is there any updates on this feature ?

**Comment by Motouom** [comment]
> Hello, is there any updates on this feature ?

hello @emouty , sorry for the late reply, yeah it will be made available in the next release.
